openquake/server/views.py

```diff
@@ -326,13 +326,8 @@ def calc_remove(request, calc_id):
         message = logs.dbcmd('del_calc', calc_id, user)
     except dbapi.NotFound:
         return HttpResponseNotFound()
-    if isinstance(message, list):  # list of removed files
-        return HttpResponse(content=json.dumps(message),
-                            content_type=JSON, status=200)
-    else:  # FIXME: the error is not passed properly to the javascript
-        logging.error(message)
-        return HttpResponse(content=message,
-                            content_type='text/plain', status=500)
+    return HttpResponse(content=json.dumps(message),
+                        content_type=JSON, status=200)
```

Comment:
If the REST request fails because calculation is owned by another user it should return a 403 instead (IMHO):

```
403 - Forbidden

The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated. If the request method was not HEAD and the server wishes to make public why the request has not been fulfilled, it SHOULD describe the reason for the refusal in the entity.  If the server does not wish to make this information available to the client, the status code 404 (Not Found) can be used instead.
```